torchgeo/datamodules/inria.py

```diff
@@ -0,0 +1,236 @@
+            self.root_dir, split="train", transforms=train_transforms
+        )
+
+        self.train_dataset: Dataset[Any]
+        self.val_dataset: Dataset[Any]
+        self.test_dataset: Dataset[Any]
+
+        if self.val_split_pct > 0.0:
+            if self.test_split_pct > 0.0:
+                self.train_dataset, self.val_dataset, self.test_dataset = dataset_split(
+                    train_dataset,
+                    val_pct=self.val_split_pct,
+                    test_pct=self.test_split_pct,
+                )
+            else:
+                self.train_dataset, self.val_dataset = dataset_split(
+                    train_dataset, val_pct=self.val_split_pct
+                )
+                self.test_dataset = self.val_dataset
+        else:
+            self.train_dataset = train_dataset
+            self.val_dataset = train_dataset
+            self.test_dataset = train_dataset
+
+        assert self.predict_on == "test"
+        self.predict_dataset = InriaAerialImageLabeling(
+            self.root_dir, self.predict_on, transforms=test_transforms
+        )
+
+    def train_dataloader(self) -> DataLoader[Any]:
+        """Return a DataLoader for training."""
+        return DataLoader(
+            self.train_dataset,
+            batch_size=self.batch_size,
+            num_workers=self.num_workers,
+            collate_fn=collate_wrapper,
+            shuffle=True,
+        )
+
+    def val_dataloader(self) -> DataLoader[Any]:
+        """Return a DataLoader for validation."""
+        return DataLoader(
+            self.val_dataset,
+            batch_size=1,
+            num_workers=self.num_workers,
+            collate_fn=collate_wrapper,
+            shuffle=False,
+        )
+
+    def test_dataloader(self) -> DataLoader[Any]:
+        """Return a DataLoader for testing."""
+        return DataLoader(
+            self.test_dataset,
+            batch_size=1,
+            num_workers=self.num_workers,
+            collate_fn=collate_wrapper,
+            shuffle=False,
+        )
+
+    def predict_dataloader(self) -> DataLoader[Any]:
+        """Return a DataLoader for prediction."""
+        return DataLoader(
+            self.predict_dataset,
+            batch_size=1,
+            num_workers=self.num_workers,
+            collate_fn=collate_wrapper,
+            shuffle=False,
+        )
+
+    def on_after_batch_transfer(
+        self, batch: Dict[str, Any], dataloader_idx: int
+    ) -> Dict[str, Any]:
+        """Apply augmentations to batch after transferring to GPU.
+
+        Args:
+            batch (dict): A batch of data that needs to be altered or augmented.
+            dataloader_idx (int): The index of the dataloader to which the batch
+            belongs.
+
+        Returns:
+            dict: A batch of data
+        """
+        # Training
+        if self.trainer.training and self.augmentations is not None:  # type: ignore[union-attr] # noqa: E501
```

Comment:
In the other modules, we use something like:
```python
        if (                                                                             
            hasattr(self, "trainer")                                                     
            and self.trainer is not None                                                 
            and hasattr(self.trainer, "training")                                        
            and self.trainer.training                                                    
        ):              
```
to avoid mypy errors. Mypy is actually correct in this case, all of these methods may or may not be None, so it's good to check first whether they are defined.

Reply:
Makes sense